### PR TITLE
(PUP-9091) Change deprecation for top level construct to an error

### PR DESCRIPTION
**Pull request by kris-bosland**

### CODE REVIEW

**Comment by justinstoller** [comment]
In case you were worried, Server is prepared for this change as of https://github.com/puppetlabs/puppetserver/pull/1791   (:

File: spec/unit/pops/validator/validator_spec.rb [hunk 1]
```diff
@@ -149,9 +149,9 @@ def with_environment(environment, env_params = {})
     it 'produces a deprecation for illegal top level constructs' do
```
Comment (jhelwig) [inline spec/unit/pops/validator/validator_spec.rb]: Rename test? `s/a deprecation/an error/`?

File: spec/unit/pops/validator/validator_spec.rb [hunk 2]
```diff
@@ -193,9 +193,9 @@ def with_environment(environment, env_params = {})
...
```
Comment (jhelwig) [inline spec/unit/pops/validator/validator_spec.rb]: Rename test? `s/a deprecation/an error/`?

File: spec/unit/pops/validator/validator_spec.rb [hunk 3]
```diff
@@ -244,9 +244,9 @@ def with_environment(environment, env_params = {})
...
```
Comment (jhelwig) [inline spec/unit/pops/validator/validator_spec.rb]: Rename test? `s/a deprecation/an error/`?

File: spec/unit/pops/validator/validator_spec.rb [hunk 4]
```diff
@@ -274,9 +274,9 @@ def with_environment(environment, env_params = {})
...
```
Comment (jhelwig) [inline spec/unit/pops/validator/validator_spec.rb]: Rename test? `s/a deprecation/an error/`?

**Pull request merged**